akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -31,6 +32,31 @@ object Sink {
     new Sink(scaladsl.Sink.fold[U, In](zero)(f.apply).toCompletionStage())
 
   /**
+   * A `Sink` that will use Java 8 `Collector` to reduce and optionally transform
+   * accumulated result into a final representation.
+   * As collector is mutable you can use it only with one stream.
+   * Reduction operations performs sequentially.
+   */
+  def javaCollector[T, A, R](collector: java.util.stream.Collector[T, A, R]): Sink[T, CompletionStage[R]] =
+    new Sink(scaladsl.Sink.javaCollector[T, A, R](collector).toCompletionStage())
+
+  /**
+   * A `Sink` that will use Java 8 `Collector` to reduce and optionally transform
+   * accumulated result into a final representation.
+   * As collector is mutable you can use it only with one stream.
```

Comment:
Collector was as parameter object in starting point that @drewhk prototyped. Can make it factory function.